### PR TITLE
Added a basic Portable Executable linker to stage2

### PR DESCRIPTION
**Pull request by alexnask**
Currently it can only output image files.  
For object file support, we need to add a COFF symbol table to the output.  

Next step will be adding export and import tables and PDB debug information.

### CODE REVIEW

File: src-self-hosted/main.zig [hunk 1]
```diff
@@ -524,17 +524,22 @@ fn buildOutputType(
             try stderr.print("\nUnable to parse command: {}\n", .{@errorName(err)});
             continue;
         }) |line| {
-            if (mem.eql(u8, line, "update")) {
+            const actual_line = if (line[line.len - 1] == '\r')
```
Comment (andrewrk) [inline src-self-hosted/main.zig:527]: `const actual_line = std.mem.trimRight(u8, line, "\r\n ");`

**Review by andrewrk — APPROVED** [review]
Excellent! This is ready to land. You up for resolving that conflict?

**Comment by andrewrk** [comment]
You can remove the `(planned)` here :)

https://github.com/ziglang/zig/blob/88724b2a89157ecc3a8eea03aa0f8a6b66829915/src-self-hosted/main.zig#L156

**Comment by andrewrk** [comment]
Might want to run those macho changes by @kubkon - I think he was in the middle of making use of that DWARF info (macho does use DWARF it just puts the data in a slightly different place)

**Review by kubkon — APPROVED** [review]
LGTM! Disabling DWARF in MachO is absolutely fine for now as I haven’t really done any work on it just yet, and I’ll be more than fine with re-enabling it myself later on as MachO progresses. :-)

**Comment by andrewrk** [comment]
:tada:

**Pull request merged**